src/wh_crypto.c

```diff
@@ -147,4 +148,51 @@ int wh_Crypto_UpdatePrivateOnlyEccKey(ecc_key* key, uint16_t pub_size,
 }
 #endif /* HAVE_ECC */
 
+#ifdef HAVE_CURVE25519
+ int wh_Crypto_SerializeCurve25519Key(curve25519_key* key,
+        uint16_t max_size, uint8_t* buffer, uint16_t *out_size)
```

Comment:
do we need to check that the `curve25519_key` is a true keypair and not only public or private part? Also, do we want to support private/public only keys?

Reply:
Hmm.  The wc code handles either/both by using separate parts. We always assume both are present but support either being absent since the buffer is fixed side.  I think an all 0 key is not valid so it would be handled as such, but you do raise a good question.  I'll look into it.